Sources/MapboxNavigation/LaneView.swift

```diff
@@ -98,127 +236,49 @@ open class LaneView: UIView {
-                    LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                } else {
-                    // pick the color of the two parts (straight & turn) depending on which direction is being suggested
-                    let turnArrowColor: UIColor
-                    let straightArrowColor: UIColor
-
-                    switch self.maneuverDirection {
-                    case .straightAhead:
-                        straightArrowColor = appropriatePrimaryColor
-                        turnArrowColor = appropriateSecondaryColor
-                    default:
-                        straightArrowColor = appropriateSecondaryColor
-                        turnArrowColor = appropriatePrimaryColor
-                    }
-
-                    LanesStyleKit.drawLaneRightOnly(frame: bounds, resizing: resizing, primaryColor: turnArrowColor, secondaryColor: straightArrowColor, flipHorizontally: true)
-                }
-            } else if indications.description.components(separatedBy: ",").count >= 2 {
-                // Hack:
-                // Account for a configuation where there is no straight lane
-                // but there are at least 2 indications.
-                // In this situation, just draw a left/right arrow
-                if maneuverDirection == .sharpRight || maneuverDirection == .right {
-                    LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                } else if maneuverDirection == .slightRight {
-                    LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                } else {
-                    LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                }
-            } else if indications.isSuperset(of: [.sharpRight]) || indications.isSuperset(of: [.right]) || indications.isSuperset(of: [.slightRight]) {
-                if indications == .slightRight {
-                    LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                } else {
-                    LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                }
-            } else if indications.isSuperset(of: [.sharpLeft]) || indications.isSuperset(of: [.left]) || indications.isSuperset(of: [.slightLeft]) {
-                if indications == .slightLeft {
-                    LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                } else {
-                    LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                }
```

Comment:
Overall, the logic shouldn’t differ too much from what the old logic was _trying_ to do. But now we’re separating the process into discrete steps and conflating the sharp and slight turns using some shorthand extensions on LaneIndication and ManeuverDirection; these changes make it easier to spot gaps in the logic, of which there were many.